### PR TITLE
[MAINPORT] Adds an Age gate

### PR DESCRIPTION
**Pull request by buffyuwu**
## Changelog
:cl:
add: ports main's age gate system
/:cl:

near direct port of https://github.com/Citadel-Station-13/Citadel-Station-13/pull/13757

### CODE REVIEW

**Comment by timothyteakettle** [comment]
I did this on main by banning anyone who failed it, but made it harder to accidentally fail by having them input their month and year (and if its ambiguous if they're 17 or 18, then it asks for day)

**Comment by buffyuwu** [comment]
> I did this on main by banning anyone who failed it, but made it harder to accidentally fail by having them input their month and year (and if its ambiguous if they're 17 or 18, then it asks for day)

That would be super cool to have for RP! With this I want to get the ball rolling with the basic 18+ verification that all adult content providers use: the simple 'are you 18 or older' age verification before entering. I want to let this be a framework to get the ball rolling for a potentially prettier age gate in the future.

With this age gate, we can take steps to replace the archaic discord form application with an age verification system similar to what you made. Replacing the panic bunker/application process will allow for our playerbase to grow from it's stagnant low number, demonstrated very clearly from times where we had no panic bunker vs when we are using it a significant increase in population where that barrier is not present.

**Comment by Captain277** [comment]
Forgot this one was still up. Just gonna close this PR as post-testing period, community results regarding lifting the Panic Bunker were firmly in the negative.

File: code/modules/mob/new_player/new_player.dm [hunk 1]
```diff
@@ -24,7 +24,23 @@
...
 
+
+/mob/new_player/proc/verifyage()
+	if(client) // sanity
+		var/alert = alert(src, "This is an 18+ roleplay server.\n(You must confirm that you are 18+ to proceed.)", "Age Verification", "I am 18+", "I am not 18+") 	
+		if(alert != "I am 18+")
+			var/text = "[key_name(usr)] failed initial age verification."
+			message_admins(text)
+			log_admin(text)
+			Logout()
```
Comment (LetterN) [inline code/modules/mob/new_player/new_player.dm:35]: ban them?

**Review by LetterN — CHANGES_REQUESTED** [review]

**Comment by buffyuwu** [comment]
re-opened amid maint discussions

**Review by LetterN — APPROVED** [review]
code good

**Pull request merged**